From f82f6f5349987d609a9d0eae8170c76d81c9bf32 Mon Sep 17 00:00:00 2001
From: Arder_Chen <Arder_Chen@phoenix.com>
Date: Thu, 28 Apr 2022 15:26:36 +0800
Subject: [PATCH] [Brightoncity] Fixed [BRIG-157] [IPMI] Some sensors are shown
 in WEB but not shown in IPMI sensor list.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Related issue: [BRIG-157] [IPMI] Some sensors are shown in WEB but not shown in IPMI sensor list.

Description:
1. After added the solution of BRIG-40(commit 0efe7f1), CPU power and total power sensor shown in WEB.
2. Memory power sensor already removed(commit ce1a81c) but still shown in WEB.
   a. Check the service for node manager:
      root@brightoncity:~# busctl tree xyz.openbmc_project.NodeManagerProxy
		└─/xyz
		  └─/xyz/openbmc_project
			├─/xyz/openbmc_project/NodeManager
			│ └─/xyz/openbmc_project/NodeManager/Domain
			│   └─/xyz/openbmc_project/NodeManager/Domain/5
			├─/xyz/openbmc_project/Power
			│ └─/xyz/openbmc_project/Power/PowerMetric
			├─/xyz/openbmc_project/sensors
			│ └─/xyz/openbmc_project/sensors/power
			│   ├─/xyz/openbmc_project/sensors/power/CPU_Power
			│   ├─/xyz/openbmc_project/sensors/power/Memory_Power	<==
			│   └─/xyz/openbmc_project/sensors/power/total_power
			├─/xyz/openbmc_project/software
			│ └─/xyz/openbmc_project/software/me
			└─/xyz/openbmc_project/status
			  └─/xyz/openbmc_project/status/me
   b. Remove Memory Power config in createSensors() function for NodeManagerProxy.cpp.
   c. Check the service for node manager:
      root@brightoncity:~# busctl tree xyz.openbmc_project.NodeManagerProxy
		└─/xyz
		  └─/xyz/openbmc_project
			├─/xyz/openbmc_project/NodeManager
			│ └─/xyz/openbmc_project/NodeManager/Domain
			│   └─/xyz/openbmc_project/NodeManager/Domain/5
			├─/xyz/openbmc_project/Power
			│ └─/xyz/openbmc_project/Power/PowerMetric
			├─/xyz/openbmc_project/sensors
			│ └─/xyz/openbmc_project/sensors/power
			│   ├─/xyz/openbmc_project/sensors/power/CPU_Power
			│   └─/xyz/openbmc_project/sensors/power/total_power
			├─/xyz/openbmc_project/software
			│ └─/xyz/openbmc_project/software/me
			└─/xyz/openbmc_project/status
			  └─/xyz/openbmc_project/status/me

Test Step:
1. Login WEB and check these sensors if exist in "Health->Sensors" page. It should exist.
---
 NodeManagerProxy.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/NodeManagerProxy.cpp b/NodeManagerProxy.cpp
index 3502e6a..bf78af8 100644
--- a/NodeManagerProxy.cpp
+++ b/NodeManagerProxy.cpp
@@ -126,9 +126,11 @@ void createSensors()
     configuredSensors.push_back(
         std::make_unique<GlobalPowerCpu>(server, 0, 510, "power", "CPU_Power",
                                          globalPowerStats, cpuSubsystem, 0));
+    #if 0
     configuredSensors.push_back(std::make_unique<GlobalPowerMemory>(
         server, 0, 255, "power", "Memory_Power", globalPowerStats,
         memorySubsystem, 0));
+    #endif
 }
 
 void createAssociations()
